### ISSUE TITLE
Could not find page count

### ISSUE THREAD

**Issue by DancingPotato30** [Lofter1/anyflip-downloader#19]
PDF download doesn't happen when I download from this like "https://anyflip.com/npkza/fxkb" and seems to be only this link. An error shows up on the console, stating the date, then the time, then "Could not find page count". Example: "2023/10/13 12:49:35 could not find page count". Other PDFs work just fine.

OS: Windows

Use anyflip-downloader with this link "https://anyflip.com/npkza/fxkb", flags don't change behaviour.

Expected behavior:
For it to download the pdf as normal. 

Notes:
The tool was downloaded by the command, not the go installer

**Comment by LukeKilmartin** [comment]
Same issue here, using https://anyflip.com/ydqsm/ybmo

**Comment by nazguleater** [comment]
Same here, https://anyflip.com/njqqe/alce

**Comment by vrailink** [comment]
I did some digging in the code and I believe i found the issue, if I may:

The application is expecting a parameter named totalPageCount in the config.js file. However, the correct parameter in the file is named pageCount.

**Solution:**

Modify the regular expression in configjs.go that searches for the page count.
Change the line:

`r := regexp.MustCompile("\"?(bookConfig\\.)?totalPageCount\"?[=:]\"?\\d+\"?")`

to:

`r := regexp.MustCompile("\"?(bookConfig\\.)?pageCount\"?[=:]\"?\\d+\"?")`

After making the change and rebuilding the application, the issue is resolved.

**Comment by Lofter1** [comment]
@vrailink thanks for the initiative. Once I have some free time I'm trying to find a solution for that. First thing I could think of is making the `total` in the name optional, as some config.js files have totalPageCount (at least, that used to be the case). However, I can see the potential for this fix to break other book downloads, so testing would be needed to determine whether that solution works or if I need to find a better solution. 
I also want to work on some debug flags, so the fix might

**Comment by vrailink** [comment]
This is my first time commenting on issues on GitHub, so if there's a more efficient way for me to assist or contribute, please let me know—I'm eager to learn and help!

You're right about ensuring compatibility. Checking for both pageCount and totalPageCount might be a good approach? 

If neither is present, then we can raise the error:

```go
func getPageCount(configjs string) (int, error) {
    r := regexp.MustCompile("\"?(bookConfig\\.)?pageCount\"?[=:]\"?\\d+\"?")
    match := r.FindString(configjs)

    if match == "" {
        r = regexp.MustCompile("\"?(bookConfig\\.)?totalPageCount\"?[=:]\"?\\d+\"?")
        match = r.FindString(configjs)
    }

    if match == "" {
        return 0, errors.New("could not find page count")
    }

    if strings.Contains(match, "=") {
        match = strings.Split(match, "=")[1]
    } else if strings.Contains(match, ":") {
        match = strings.Split(match, ":")[1]
    }

    match = strings.ReplaceAll(match, "\"", "")
    return strconv.Atoi(match)
}
```

Unfortunately, I don't have access to a link that uses the older format (totalPageCount), so it would be essential to test this.

Thanks and let me know how else I can be of assistance!

**Comment by Lofter1** [comment]
@vrailink if you want to, you can open a pull request. that way, I only would need to test the code and you would automatically be listed as a contributor to this project, giving you credit for having found a solution 🙂 
however, if you want to do that, I suggest to simply make the `total` in the regex an optional string. It's the simpler, more efficient solution and a more complex one can be used later, still, if it turns out a more complex one is needed 🙂

**Comment by vrailink** [comment]
Thanks for the guidance.
I've successfully created the pull request.

Regarding making `total` optional in the regex, I noticed that the case sensitivity due to the difference in capitalization of the letter 'p' in `PageCount` vs `pageCount` is causing an issue. We might need a more robust approach to handle both scenarios.

**Comment by Lofter1** [comment]
there is a regex for that, too. If I'm not mistaken, `[Pp]` should work. This checks whether a `P` or a `p` can be found at that point in the string. So the regex would first look for the optional `total` and then if it finds either a lower or upper case `p` and then it looks for the string `ageCount`. So the full regex should look somewhat like this `?(total)[Pp]ageCount`. But your solution also works 🙂

**Comment by vrailink** [comment]
I've updated the regex as and pushed the changes to the PR. Using `[Pp]` is indeed a more elegant solution.
Thanks for all the help! 🙂

**Comment by hessedelohim** [comment]
Hi, this solution will be applied for the other OS installer also? Regards

**Comment by L-o-ki** [comment]
I'm having this same issue, I would change the Config.js but don't know how to re-build the application- I'm in windows OS.  Any chance this is getting corrected soon/anyone have a place where I can learn how to do this?  Thanks all.

**Comment by Lofter1** [comment]
Hey everyone. Sorry, I didn't have time to test the PR and merge it, yet. I hope I have time this weekend to do so. Maybe I see if I can make up some time tonight. I know it's been a while, sorry for the delay and thanks for everyone's patience.

**Comment by Florian47** [comment]
hey everyone. Tried their suggestions and it worked flawlessly. Simply fork their project and manually make their changes. Then run go build through terminal. Thats it. I will attach zip with the exe inside. Cheers :)
[anyflip-downloader.zip](https://github.com/Lofter1/anyflip-downloader/files/13245743/anyflip-downloader.zip)

**Comment by Florian47** [comment]
and thanks to @vrailink and @Lofter1!

**Comment by hessedelohim** [comment]
> hey everyone. Tried their suggestions and it worked flawlessly. Simply fork their project and manually make their changes. Then run go build through terminal. Thats it. I will attach zip with the exe inside. Cheers :) [anyflip-downloader.zip](https://github.com/Lofter1/anyflip-downloader/files/13245743/anyflip-downloader.zip)

I use your exe with the command:
# .\anyflip-downloader.exe https://anyflip.com/xxxxx/xxxx/ 

but still get the same error:

https://anyflip.com/xxxxx/xxxx/ 
2023/11/04 10:55:08 could not find page count

Im missing some step?

**Comment by ija1998** [comment]
Hey sorry for the off-topic comment, where does the file get download to?

**Comment by Lofter1** [comment]
Fixed in #24
